File: frontend/src/services/receptionApi.ts
import { apiClient } from './apiClient';
import type { ReceptionTravaux, PhotoReception } from '../types';

export interface ReceptionTravauxRequest {
  autorisationTravailId: string;
  dateReelleDebut?: string;
  dateReelleFin?: string;
  travauxConformes: boolean;
  zoneNettoyee: boolean;
  consignationRetiree: boolean;
  equipementRemisEnService: boolean;
  installationRemiseEnEtat: boolean;
  essaisEffectues: boolean;
  essaisConformes: boolean;
  travauxRealises?: string;
  commentaires?: string;
}

export const receptionApi = {
  getByAtId: async (atId: string): Promise<ReceptionTravaux> => {
    const response = await apiClient.get<ReceptionTravaux>(`/receptions/at/${atId}`);
    return response.data;
  },

  getById: async (id: string): Promise<ReceptionTravaux> => {
    const response = await apiClient.get<ReceptionTravaux>(`/receptions/${id}`);
    return response.data;
  },

  create: async (request: ReceptionTravauxRequest): Promise<ReceptionTravaux> => {
    const response = await apiClient.post<ReceptionTravaux>('/receptions', request);
    return response.data;
  },

  update: async (id: string, request: ReceptionTravauxRequest): Promise<ReceptionTravaux> => {
    const response = await apiClient.put<ReceptionTravaux>(`/receptions/${id}`, request);
    return response.data;
  },

  signer: async (id: string, signaturePath: string): Promise<ReceptionTravaux> => {
    const response = await apiClient.put<ReceptionTravaux>(`/receptions/${id}/signer`, signaturePath, {
      headers: { 'Content-Type': 'text/plain' },
    });
    return response.data;
  },

  cloturer: async (id: string): Promise<ReceptionTravaux> => {
    const response = await apiClient.put<ReceptionTravaux>(`/receptions/${id}/cloturer`);
    return response.data;
  },

  getPhotos: async (id: string): Promise<PhotoReception[]> => {
    const response = await apiClient.get<PhotoReception[]>(`/receptions/${id}/photos`);
    return response.data;
  },

  ajouterPhoto: async (id: string, file: File, description?: string): Promise<PhotoReception> => {
    const formData = new FormData();
    formData.append('file', file);
    if (description) formData.append('description', description);
    const response = await apiClient.post<PhotoReception>(`/receptions/${id}/photos`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  },
};

File: mobile/lib/features/at/data/reception_api.dart
/// Client API pour la réception des travaux (GET /receptions/at/{atId}, POST /receptions, etc.)
library;

import '../../../core/network/api_client.dart';
import 'models/reception_travaux.dart';

class ReceptionApi {
  final ApiClient _client;
  const ReceptionApi(this._client);

  Future<ReceptionTravaux?> getByAtId(String atId) async {
    try {
      final res = await _client.get<Map<String, dynamic>>('/receptions/at/$atId');
      if (res.data == null) return null;
      return ReceptionTravaux.fromJson(res.data!);
    } catch (_) {
      return null;
    }
  }

  Future<ReceptionTravaux> getById(String id) async {
    final res = await _client.get<Map<String, dynamic>>('/receptions/$id');
    return ReceptionTravaux.fromJson(res.data!);
  }

  Future<ReceptionTravaux> create(ReceptionTravaux reception) async {
    final res = await _client.post<Map<String, dynamic>>(
      '/receptions',
      body: reception.toJson(),
    );
    return ReceptionTravaux.fromJson(res.data!);
  }

  Future<ReceptionTravaux> update(String id, ReceptionTravaux reception) async {
    final res = await _client.put<Map<String, dynamic>>(
      '/receptions/$id',
      body: reception.toJson(),
    );
    return ReceptionTravaux.fromJson(res.data!);
  }

  Future<ReceptionTravaux> signer(String id, String signaturePath) async {
    final res = await _client.put<Map<String, dynamic>>(
      '/receptions/$id/signer',
      body: signaturePath,
    );
    return ReceptionTravaux.fromJson(res.data!);
  }

  Future<ReceptionTravaux> cloturer(String id) async {
    final res = await _client.put<Map<String, dynamic>>('/receptions/$id/cloturer');
    return ReceptionTravaux.fromJson(res.data!);
  }
}
